src/StackExchange.Redis/Message.cs

```diff
@@ -621,14 +630,58 @@ internal bool ComputeResult(PhysicalConnection connection, in RawResult result)
             }
         }
 
+        internal void OverrideConnectionRestoreFlagIfNotSet(OnConnectionRestore? onConnectionRestore)
+        {
+            if (onConnectionRestore.HasValue && !IsOnConnectionRestoreFlagSet)
+            {
+                switch (onConnectionRestore)
+                {
+                    case OnConnectionRestore.AlwaysRetry:
+                        Flags |= CommandFlags.OnConnectionRestoreAlwaysRetry;
+                        break;
+                    case OnConnectionRestore.RetryIfNotYetSent:
+                        Flags |= CommandFlags.OnConnectionRestoreRetryIfNotYetSent;
+                        break;
+                    case OnConnectionRestore.NoRetry:
+                    default:
+                        Flags |= CommandFlags.OnConnectionRestoreNoRetry;
+                        break;
+                }
```

Comment:
Can switch expression here, though the bigger win would be to unify the enum and eliminate the conversion.

Reply:
I found there are few internal teams that are using commandflags per request. So I went with Marc's suggestion to have commandretry as commandflag with a validation to allow retry options on it.